packages/react-server-cli/src/buildWebpackConfigs.js

```diff
@@ -0,0 +1,355 @@
+			chunkFilename: `[id]${longTermCaching ? ".[chunkhash]" : ""}.bundle.js`,
+		},
+		plugins: [
+			...commonWebpackConfig.plugins,
+			/*
+			TODO: need to figure out why this isn't working properly.  This file is only generated when
+	    		compiling using "hot" setting and isn't used anywhere.
+			new ChunkManifestPlugin({
+				filename: "chunk-manifest.json",
+				manifestVariable: "webpackManifest",
+			}),
+			*/
+			new webpack.optimize.CommonsChunkPlugin({
+				name:"common",
+			}),
+			new webpack.DefinePlugin({
+				REACT_SERVER_CLIENT_SIDE: 'true',
+			}),
+		],
+	});
+
+	if (minify) {
+		clientWebpackConfig.plugins.push(
+			// TODO: should this be done as babel plugin?
+			new webpack.optimize.UglifyJsPlugin(),
+		);
+	} else {
+		clientWebpackConfig.plugins.push(new webpack.SourceMapDevToolPlugin());
+	}
+
+	return clientWebpackConfig;
+}
+
+function packageCodeForNode(commonWebpackConfig, entryPoints, outputDir) {
+	// This function is required because sometimes modules are loaded like 'require("react-dom/server/index.js");' and
+	// it forces Webpack to load the whole react-dom module.
+	// https://github.com/webpack/webpack/issues/839#issuecomment-177219660
+	const nodeModulesTransform = function(context, request, callback) {
+		let absoluteRequest;
+		if (/^\./.test(request)) {
+			// this is a relative pathname
+			absoluteRequest = path.join(context, request);
+		} else {
+			absoluteRequest = request;
+		}
+		// Make all of the requests relative to node_modules for determining nested modules
+		const relativeRequest = path.normalize(absoluteRequest).replace(/^.*?\/node_modules\//, '');
+
+		// search for a '/' indicating a nested module
+		const slashIndex = relativeRequest.indexOf("/");
+		let rootModuleName;
+		if (slashIndex === -1) {
+			rootModuleName = relativeRequest;
+		} else {
+			rootModuleName = relativeRequest.substr(0, slashIndex);
+		}
+
+		// Match for root modules that are in our node_modules except Webpack.
+		// Because Webpack uses relative pathnames for itself, let's just bundle Webpack
+		if (rootModuleName !== "webpack" && nodeModules.hasOwnProperty(rootModuleName)) {
+			// This is in our node_modules, so we can safely list it as external
+			callback(null, "commonjs " + request);
+		} else {
+			// This is NOT in our node_modules, so we must bundle it.
+			callback();
+		}
+	};
+
+	let serverWebpackConfig = Object.assign({}, commonWebpackConfig, {
+		target: "node",
+		node: {
+			__dirname  : false,
+			__filename : false,
+		},
+		entry: entryPoints,
+		externals: nodeModulesTransform,
+		output: {
+			path: outputDir,
+			filename: '[name].bundle.js',
+			chunkFilename: '[id].bundle.js',
+			libraryTarget: 'commonjs2',
+			pathinfo: true,
+		},
+		devtool: 'source-map',
+		plugins: [
+			...commonWebpackConfig.plugins,
+			new webpack.SourceMapDevToolPlugin(),
+			new webpack.NormalModuleReplacementPlugin(/\.(css|less|sass|scss)$/, 'node-noop'),
```

Comment:
This is a well intended performance improvement, but it breaks CSS modules. In order to determine CSS class names, we must compile all CSS files for the server code as well. We just need to omit the style-loader because it depends on the DOM.